k-distribution/k-tutorial/1_basic/15_configurations/README.md

```diff
@@ -242,9 +242,9 @@ of this term in later lessons.
 
 1. Modify the definition from the previous exercise in this lesson so that the
 Boolean cell you created is initialized to false. Then add a production
-`syntax Stmt ::= Bool ";" Exp`, and a rule that uses this `Stmt` to set the
+`syntax Stmt ::= Bool ";" Stmt | ""`, and a rule that uses this `Stmt` to set the
```

Comment:
Productions that don't contain any syntax `""` are very tricky for the inner parser to handle. So they are forbidden.
You can get degenerate things like `rule =>` because you can match nothing, anywhere.
You can use them in your program language but using a separate module that is not included in the main semantics module. This is a tricky thing, and I don't think it's worth adding to the basic lessons.

Reply:
Good to know. I just wanted to make the exercise easier for testing the actual solution by enabling a `reset` after a `true`. We can change it to something like

```k
  syntax Stmt ::= Bool ";" Stmt | Exp
  syntax Stmt ::= "reset" ";" Stmt
```

So the sequence could be `-cPGM='true;reset;1`, for example.

Or I might have misunderstood about the exercise myself :-)

Reply:
I think I understand it in the same way. But making it explicit may make it a bit too easy.
Leaving the `Stmt` sequencing out makes the user think about more than just copy/pasting something in.
So I think it is fine as it was.

Reply:
Hmm... Why not, I mean one can figure it out how to test it.... But personally I would stick to making exercises ask a single thing ~> in this case, do I understand how to write the semantics for reset by actually referencing the initial value instead of copy&pasting it/hardwiring it twice? But sure, we can keep it this way.